content/en/blog/_posts/2022-12-05-Security-Behavior-Analysis/index.md

```diff
@@ -0,0 +1,79 @@
+
+More specifically, let’s assume the service is vulnerable to an SQL injection, the developer failed to do proper sanitization of user input, allowing clients to send values that will change the intended behavior. In our example, if a client sends a query string with key “username” and value of _“tom or 1=1”_, the client will receive the data of all users. Exploiting this vulnerability requires the client to send an irregular string as the value. Note that naive users will not be sending a string with spaces or with the equal sign character as a username, instead they will normally send legal usernames which for example may be defined as a short sequence of characters a-z. No legal username may trigger service unplanned behavior.
+
+In this simple example we can already identify several opportunities to detect and block an attempt to exploit the vulnerability (un)intentionally left behind by the developer, making the vulnerability unexploitable. First, the malicious client behavior differs from the behavior of naive clients, as it sends irregular requests. If such change in behavior is detected and blocked, the exploit will never reach the service. Second, the service behavior in response to the exploit, differs from the service behavior in response to a regular request. The service irregular behavior may include making subsequent irregular calls to other services such as a data store, taking irregular time to respond, and/or responding to the malicious client with an irregular response (for example, containing much more data then normally sent in case of naive clients making regular requests). Service behavioral changes, if detected, will also allow blocking the exploit in different stages of the exploitation attempt.
+
+More generally:
+
+- Monitoring the behavior of clients can help detect and block exploits against service API vulnerabilities. In fact, deploying an efficient client behavior monitoring makes many vulnerabilities unexploitable and others very hard to achieve. To succeed, the offender needs to create an exploit undetectable from regular naïve requests.
+
+- Monitoring the behavior of services can help detect services as they are being exploited regardless of the attack vector used. Efficient service behavior monitoring limits what an attacker may be able to achieve as the offender need to ensure the service behavior is undetectable from regular service behavior.
+
+Combining both approaches may add a protection layer to the deployed vulnerable services, drastically decreasing the probability for anyone to successfully exploit any of the deployed vulnerable services. Next, lets identify four use cases where you need to use security-behavior monitoring.
+
+## Use cases
+One can identify the following four different stages in the life of a any service from a security standpoint. In each stage, security-behavior monitoring is required to meet different challenges:
```

Comment:
Nit: I am curious on why "Security-Behavior" is used instead of "security behavior" in the blog?

Reply:
*"Security-Behavior"*  is a name given to "the behavior of (clients and services) from a security standpoint".
(It is also indicated at the beginning of the post)

if we just write "security behavior" - I think it throws the reader to think we are discussing the behavior of security which I think is very confusing.